### PR TITLE
fix pd_control: not rely on error text

### PR DESCRIPTION
**Pull request by weekface**
- just return an error if delete store failed
- use `GetEvictLeaderSchedulers` instead

@tennix @xiaojingchen PTAL

### CODE REVIEW

File: pkg/controller/pd_control.go [hunk 1]
```diff
@@ -407,15 +396,24 @@ func (pc *pdClient) BeginEvictLeader(storeID uint64) error {
 	if res.StatusCode == http.StatusOK {
 		return nil
 	}
-	err2 := readErrorBody(res.Body)
-	if strings.Contains(err2.Error(), schedulerExisted) {
-		return nil
+
+	evictLeaderSchedulers, err := pc.GetEvictLeaderSchedulers()
```
Comment (xiaojingchen) [inline pkg/controller/pd_control.go]: this should be exec before create evict leader, if the scheduler leader exists, then not need to post sheduler leader

File: pkg/controller/pd_control.go [hunk 2]
```diff
@@ -428,11 +426,19 @@ func (pc *pdClient) EndEvictLeader(storeID uint64) error {
 	if res.StatusCode == http.StatusOK || res.StatusCode == http.StatusNotFound {
 		return nil
 	}
+
 	err2 := readErrorBody(res.Body)
-	if strings.Contains(err2.Error(), schedulerNotFound) {
-		return nil
+	evictLeaderSchedulers, err := pc.GetEvictLeaderSchedulers()
```
Comment (xiaojingchen) [inline pkg/controller/pd_control.go]: ditto

File: pkg/controller/pd_control.go [hunk 3]
```diff
@@ -328,13 +324,6 @@ func (pc *pdClient) DeleteStore(storeID uint64) error {
 		return err
 	}
 
-	// FIXME: We should not rely on error text
-	// Remove a tombstone store should returns "store has been removed"
-	bodyStr := string(body)
-	if strings.Contains(bodyStr, "store has been removed") {
-		return nil
-	}
-
```
Comment (gregwebs) [inline pkg/controller/pd_control.go]: PD now actually returns a structured json response with an error code and an error code as a header.

Comment (weekface) [reply pkg/controller/pd_control.go]: Yes, but the older version don't have the error code and header.

Return the error is ok for this situation. The `DeleteStore` function is reentrant.

**Pull request merged**